Gems/Atom/RPI/Code/Tests.Builders/AnyAssetBuilderTest.cpp

```diff
@@ -239,26 +240,44 @@ namespace UnitTest
             stream->Open(assetFile, 0, fileLength);
             stream->BlockUntilLoadComplete();
             m_assetHandler->LoadAssetData(outAsset, stream, {});
+            stream->Close();
+
+            // Force a file streamer flush to ensure that file handles don't remain used
+            auto streamer = AZ::Interface<AZ::IO::IStreamer>::Get();
+            AZStd::binary_semaphore wait;
+            AZ::IO::FileRequestPtr flushRequest = streamer->FlushCaches();
+            streamer->SetRequestCompleteCallback(flushRequest, [&wait]([[maybe_unused]] AZ::IO::FileRequestHandle request)
+                {
+                    wait.release();
+                });
+            streamer->QueueRequest(flushRequest);
+            wait.acquire();
+
             return outAsset;
         }
         
     };
     
     TEST_F(AnyAssetBuilderTests, ProcessJobBasic)
     {
+        const char* testAssetName = "AnyAssetTest.source";
+        AZ::Test::ScopedAutoTempDirectory productDir;
+        AZ::Test::ScopedAutoTempDirectory sourceDir;
+        AZStd::string sourceFilePath;
+        AzFramework::StringFunc::Path::Join(sourceDir.GetDirectory(), testAssetName, sourceFilePath, true, true);
+
         // Basic test: test data before and after are same. Test data class doesn't have converter or asset reference.
         RPI::AnyAssetBuilder builder;
         AssetBuilderSDK::ProcessJobRequest request;
         AssetBuilderSDK::ProcessJobResponse response;
 
         // Initial job request
-        const char* testAssetName = "AnyAssetTest.source";
-        request.m_fullPath = testAssetName;
-        request.m_tempDirPath = m_currentDir;
+        request.m_fullPath = sourceFilePath;
```

Comment:
This can use `AZ::IO::Path.Native` to retrieve the underlying string member from the AZ::IO::Path instead of `.c_str()`
```suggestion
        request.m_fullPath = sourceFilePath.Native();
```